casper/src/test/scala/coop/rchain/casper/BlockRandomSeedSpec.scala

```diff
@@ -0,0 +1,90 @@
+package coop.rchain.casper
+
+import cats.syntax.all._
+import coop.rchain.crypto.PublicKey
+import coop.rchain.rspace.hashing.Blake2b256Hash
+import coop.rchain.shared.Base16
+import coop.rchain.shared.scalatestcontrib.convertToAnyShouldWrapper
+import coop.rchain.store.InMemoryKeyValueStore
+import monix.eval.Task
+import monix.execution.Scheduler.Implicits.global
+import org.scalacheck.Prop.AnyOperators
+import org.scalacheck.{Arbitrary, Gen}
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatestplus.scalacheck.Checkers
+import scodec.bits.ByteVector
+
+class BlockRandomSeedSpec extends AnyFlatSpec with Checkers {
+  "encoding coop.rchain.casper.BlockRandomSeed object and decoding it" should "give original object" in {
+    val blockRandomSeedGen = for {
+      shardId                <- Gen.stringOfN(6, Gen.alphaChar)
+      blockNumber            <- Gen.choose(0L, 1000000L)
+      validatorPublicKeyList <- Gen.listOfN(120, Gen.hexChar)
+      validatorPublicKey     = PublicKey(validatorPublicKeyList.map(_.toByte).toArray)
+      hash                   <- Gen.listOfN(Blake2b256Hash.length, Gen.hexChar)
+      preStateHash           = Blake2b256Hash.fromByteArray(hash.map(_.toByte).toArray)
+
+    } yield BlockRandomSeed(shardId, blockNumber, validatorPublicKey, preStateHash)
+
+    implicit lazy val blockRandomSeedArbitrary = Arbitrary(blockRandomSeedGen)
+
+    check { blockRandomSeed: BlockRandomSeed =>
+      val serialized    = BlockRandomSeed.codecBlockRandomSeed.encode(blockRandomSeed).require
+      val reconstructed = BlockRandomSeed.codecBlockRandomSeed.decode(serialized).require.value
+
+      reconstructed ?= blockRandomSeed
+    }
+  }
+
+  "encoding Blake2b256Hash object and decoding it" should "give original object" in {
+    val hashGen = for {
+      hashValue <- Gen.listOfN(Blake2b256Hash.length, Gen.hexChar)
+    } yield Blake2b256Hash.fromByteArray(hashValue.map(_.toByte).toArray)
+
+    implicit lazy val blockRandomSeedArbitrary = Arbitrary(hashGen)
+
+    check { hash: Blake2b256Hash =>
+      val serialized = Blake2b256Hash.codecBlake2b256Hash.encode(hash).require
+      val restored   = Blake2b256Hash.codecBlake2b256Hash.decode(serialized).require.value
+
+      restored ?= hash
+    }
+  }
+
+  "generate random seed from constant BlockRandomSeed object" should "give expected value" in {
+    val validatorPublicKey = PublicKey(
+      Base16.unsafeDecode(
```

Comment:
We have extension methods to convert between string hex and ByteString or Array[Byte], ByteVector has built-in.